frontend/src/v5/ui/controls/button/button.component.tsx

```diff
@@ -18,20 +18,21 @@
 import React from 'react';
 import MuiButton, { ButtonProps } from '@material-ui/core/Button';
 import { Typography } from '@/v5/ui/controls/typography';
+import { ButtonTypeMap } from '@material-ui/core/Button/Button';
 import { LabelButton } from './button.styles';
 
 type ButtonVariants = ButtonProps['variant'] | 'label' | 'label-outlined';
 
-type IButton = Omit<ButtonProps, 'variant'> & {
+type IButton<T extends React.ElementType = ButtonTypeMap['defaultComponent']> = Omit<ButtonProps<T>, 'variant'> & {
```

Comment:
Just a  couple of questions here, not something to change.
- Why has this changed?
- is there any documentation about it?